client/sftp.js

```diff
@@ -0,0 +1,83 @@
+import sftp from 'ssh2-sftp-client';
+
+export class SFTPClient {
+  constructor(timeout, logger, encoding) {
+    this.encoding = encoding;
+    this.timeout = timeout;
+    this.log = logger;
+
+    this.readOptions = {
+      readStreamOptions: {
+        flags: 'r',
+        encoding: this.encoding,
+      },
+      pipeOptions: {
+        end: false
+      }
+    };
+
+    this.client = new sftp();
+
+    this.connectionClosed = true;
+
+    this.client.on('close', this.closedConnection);
+    this.client.on('end', this.closedConnection);
+  }
+
+  async size(path) {
+    if (this.connectionClosed) throw Exception('Connect first in order to use this function');
+
+    this.log(`Getting size for file ${path}`);
+
+    return this.connection
+      .then(() => {
+        return this.client.stat(path);
+      })
+      .then((data) => {
+        this.log(`File size ${data.size}`);
+        return data.size;
+      })
+      .catch((err) => {
+        console.error(err.message);
+      });
+  }
+
+  async access(options = {}) {
+    // ssh2-sftp-client requires username instead of user.
+    options.username = options.user;
+    options.readyTimeout = this.timeout;
+
+    this.log('Setting up sftp connection');
+    this.connection = this.client.connect(options);
```

Comment:
As above, I think we should be using `await` here.

Reply:
I didn't want to fixed that pattern that it is going to be used. The implementation calls the awaits, this works with promises just to keep it also consistent with the ftp library

Reply:
The `await` used by the implementation is waiting for the function this line is in to be finished, but because that function doesn't use `await` on this line it means that that function returns before the connection has occurred. You definitely should use an `await` on this line.